src/Sfa.Tl.Matching.Web/Controllers/ProviderController.cs

```diff
@@ -131,22 +137,40 @@ private async Task<SearchViewModel> GetSearchResultsAsync(SearchParametersViewMo
                 {
                     Results = _mapper.Map<List<SearchResultsViewModelItem>>(searchResults)
                 },
-                SearchParameters = GetSearchParametersViewModelAsync(viewModel.SelectedRouteId, viewModel.Postcode?.Trim(), viewModel.SearchRadius)
+                SearchParameters = GetSearchParametersViewModelAsync(viewModel)
             };
 
+            if (viewModel.OpportunityId == 0)
+                return resultsViewModel;
+
+            var selectedResultsViewModel = PopulateSelectedSearchResults(viewModel.OpportunityId, resultsViewModel);
+
+            return selectedResultsViewModel;
+        }
+
+        private SearchViewModel PopulateSelectedSearchResults(int opportunityId, SearchViewModel resultsViewModel)
```

Comment:
opportunityId is already set in SearchParametersViewModel.OpportunityId

Reply:
Correct. Ive removed the OpportunityId parameter